### PR TITLE
DOC: Added examples to pd.Index.get_loc

### PR DESCRIPTION
**Pull request by topper-123**
This pull request adds some examples to ``Index.get_loc`` (actually to ``Index._index_shared_docs['get_loc']``) and clarifies the return value.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -2450,7 +2450,19 @@ def _get_unique_index(self, dropna=False):
...
         -------
-        loc : int if unique index, possibly slice or mask if not
+        loc : int if unique index, slice if monotonic index, else mask
+
+        Examples
+        ---------
+        >>> unique_index = pd.Index(list('abc'))
+        >>> unique_index.get_loc('b')
+        1
+        >>> monotonic_index = pd.Index(list('abbc'))
```
Comment (jreback) [inline pandas/core/indexes/base.py]: can you add a blank line between cases

**Comment by jreback** [comment]
lgtm. ping push / green.

**Comment by topper-123** [comment]
ping

**Comment by topper-123** [comment]
@jreback Is this ok? 

(More of a general question): I force-pushed the change, is that ok or is it considered "bad form"?

**Comment by jreback** [comment]
@topper-123 force pushing to your own branch is just fine and in-fact necessary after rebasing.

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -2450,10 +2450,24 @@ def _get_unique_index(self, dropna=False):
 
         Returns
         -------
-        loc : int if unique index, possibly slice or mask if not
+        loc : int if unique index, slice if monotonic index, else mask
+
+        Examples
+        ---------
+        >>> unique_index = pd.%(target_klass)s(list('abc'))
```
Comment (jreback) [inline pandas/core/indexes/base.py]: hmm these are only valid for Index/CategoricalIndex. Do these examples show up for other indexes (IIR we are using this combined doc-string for all sub-classes). You'll have to do this as an additional ``@Appender`` to those classes only.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (topper-123) [reply pandas/core/indexes/base.py]: I think I will do this all over again. Many of the ``.get_loc`` methods are subtlely different, so it's difficult to have a common doc-string.

Before I do that, could you please explain how ``IntervalIndex.get_loc`` gets its doc-string? From the code it seems it should have one, but it does....

Comment (jreback) [reply pandas/core/indexes/base.py]: hmm, I think its inheriting it from Index.get_loc, instead this should use the appender method (guess it was omitted somehow).

**Comment by topper-123** [comment]
Hi @jreback. I've updated the examples for other index classes as well, but I've messed up the commits, lots of other people's commit are included also.

Should I just delete this branch and make a new pull request?

**Comment by jreback** [comment]
@topper-123 generally just push (or force push) to the same branch is easiest, but since you did it ok

**Pull request closed without merging**